FRAuthenticator/FRAuthenticator/Manager/AuthenticatorManager.swift

```diff
@@ -39,36 +44,61 @@ struct AuthenticatorManager {
     func createMechanismFromUri(uri: URL, onSuccess: @escaping MechanismCallback, onError: @escaping ErrorCallback) {
         
         let authType = uri.getAuthType()
-        FRALog.v("Received URI (\(uri.absoluteString)) with authType (\(authType.rawValue)), and proceeding to construct Mechanism object")
-
-        if authType == .hotp {
-            do {
-                let mechanism = try self.storeOathQRCode(uri: uri)
-                onSuccess(mechanism)
-            }
-            catch {
-                onError(error)
+        if let scheme = uri.scheme, scheme == "mfauth" {
+            FRALog.v("Evaluating policies for the new Account")
+            let result = self.policyEvaluator.evaluate(uri: uri)
+            if(!result.comply) {
+                onError(AccountError.failToRegisterPolicyViolation(result.nonCompliancePolicy!.name))
```

Comment:
Can we avoid force unwrapping just to be safe? ?? default value would be preferred